src/libraries/System.Net.Http/src/PACKAGE.md

```diff
@@ -0,0 +1,6 @@
+## About
+
+The package is obsolete and not necessary. It should NOT be used. If you use it anyway (e.g. due to dependencies out of your control), it won't cause you any harm.
+
+It is a result of failed attempt from 2016 to deliver value to .NET Framework customers outside of official .NET Framework updates. Later we realized it causes more troubles we didn't foresee, therefore we stopped shipping it for .NET Core entirely (the package will just forward to the in-box implementation) and on .NET Framework we made it exact copy of in-box code. As a result, using it does not bring any value, but does not cause any harm either.
```

Comment:
> and on .NET Framework we made it exact copy of in-box code

The package is forwarded to the in-box implementation on .NET Framework just like it is in .NET Core. If it was not the case, using the package would cause harm since it would disable in-box servicing.

Reply:
I looked into package contents, and i see it has implementation assembly for `net46` TFM, with type definitions in it, not type forwards. Is the forwarding logic hardcoded in NuGet so these assemblies aren't actually used?

Reply:
Yes, these .NET Standard 1.x assemblies are special-cased by .NET SDK/NuGet and .NET Framework loader.

- For projects that target .NET Framework `net472`+ and reference `System.Net.Http` nuget package, System.Net.Http.dll won't even show up in the project output.
- For projects that target older .NET Framework version and reference `System.Net.Http` nuget package, `System.Net.Http.dll` is going to show up in the project output but it won't be actually used at runtime if you run it on .NET Framework 4.7.2 or newer, and in some cases on earlier .NET Framework versions as well.

It means that the binary bundled in nuget package matters only if you are stuck in 2016 and still depend on the features that were delivered via this package as OOB. The number of people in this situation should be very small. I do not think that it makes sense to explain these complicated details for this corner case.